Utilities/src/test/java/edu/stanford/hivdb/mutations/MutationTest.java

```diff
@@ -538,4 +540,100 @@ public void testGenePosition() {
+		assertTrue(stopMut.hasStop());
+	}
+	
+	@Test
+	public void testIsUnusual() {
+		final Mutation unusualMut = new Mutation(Gene.RT, 1, "A");
+		final Mutation usualMut = new Mutation(Gene.RT, 1, "H");
+		final Mutation usualMuts = new Mutation(Gene.RT, 1, "HLPST");
+		final Mutation unusualMuts = new Mutation(Gene.RT, 1, "ACDEFG");
+		final Mutation mixedMuts = new Mutation(Gene.PR, 75, "AILMVSTY");
+		assertFalse(usualMut.isUnusual());
+		assertFalse(usualMuts.isUnusual());
+		assertTrue(unusualMut.isUnusual());
+		assertTrue(unusualMuts.isUnusual());
+		assertTrue(mixedMuts.isUnusual());
+			
+//		The JSON suggests so, but are some insertions and deletions 'usual?' 
+//		Arrays.asList(Gene.values()).forEach(gene -> {
+//			IntStream.range(1, gene.getLength() + 1).forEach(pos -> {
+//				System.out.println("del Gene:" + gene + " : " + pos + " " + new Mutation(gene, pos, "-").isUnusual());
+//				System.out.println("ins Gene:" + gene + " : " + pos + " " + new Mutation(gene, pos, "_").isUnusual());
+//				assertTrue(new Mutation(gene, pos, "-").isUnusual());
+//				assert(new Mutation(gene, pos, "_").isUnusual());
+//			});
+//		});
+	}
+	
+	@Test
+	public void testIsSDRM() {
+		final Mutation mut = new Mutation(Gene.PR, 24, "N");
+		final Mutation sdrmMut = new Mutation(Gene.PR, 24, "I");
+		final Mutation mixedMuts = new Mutation(Gene.RT, 230, "LI");
+		assertFalse(mut.isSDRM());
+		assertTrue(sdrmMut.isSDRM());
+		assertTrue(mixedMuts.isSDRM());
+	}
+		
+	@Test
+	public void testIsAmbiguous() {
+		final Mutation mut = new Mutation(Gene.PR, 24, "N");
+		final Mutation xMut = new Mutation(Gene.PR, 24, "X");
+		final Mutation tripMut = new Mutation(Gene.PR, 24, "N", "AAC");
+		final Mutation tripMutX = new Mutation(Gene.PR, 24, "X", "AAC");
+		final Mutation bTripMut = new Mutation(Gene.PR, 24, "N", "AAB");
+		final Mutation dTripMut = new Mutation(Gene.PR, 24, "N", "DAC");
+		final Mutation hTripMut = new Mutation(Gene.PR, 24, "S", "THT");
+		final Mutation vTripMut = new Mutation(Gene.PR, 24, "S", "TCV");
+		final Mutation nTripMut = new Mutation(Gene.PR, 24, "S", "TNA");
+		assertFalse(mut.isAmbiguous());
+		assertTrue(xMut.isAmbiguous());
+		assertFalse(tripMut.isAmbiguous());
+		assertTrue(tripMutX.isAmbiguous());
+		assertTrue(bTripMut.isAmbiguous());
+		assertTrue(dTripMut.isAmbiguous());
+		assertTrue(hTripMut.isAmbiguous());
+		assertTrue(vTripMut.isAmbiguous());
+		assertTrue(nTripMut.isAmbiguous());
+	}
+	
+	@Test
+	public void testGetHighestMutPrevalance() {
+		final Mutation prevMut = new Mutation(Gene.IN, 45, "G");
+		final Mutation prevMuts = new Mutation(Gene.IN, 45, "HKQ");
+		final Mutation prevMutZero = new Mutation(Gene.IN, 45, "C");
+		final Mutation prevMutsZero = new Mutation(Gene.IN, 45, "CDEFH");
+		final Mutation prevMutsWCons = new Mutation(Gene.IN, 45, "LHKQ");
+		final Mutation prevMutsWConsAndStop = new Mutation(Gene.IN, 45, "*LHKQ");
+		assertEquals(0.052, prevMut.getHighestMutPrevalence(), 0.0);
```

Comment:
These assertions will be expected to fail after we uploaded a new version of prevalence data derived from our core database.

Can we add a comment before this line to indicate that this situation is expected and developer had to fix the assertions one by one manually? In addition, other situations should be considered as software regressions unless the prevalence tables are just updated.

Reply:
Done. 

Can you please explain what you mean by "other situations should be considered as software regressions unless the prevalence tables are just updated"? Should I add tests beyond hard-coded prevalence checks? 

Also, since prevalence data will change periodically, would it be worthwhile to write tests that read and test against the prevalence file directly?

Reply:
Well done. Other situations, for example a change made to `Mutation` class that breaks `getHighestMutPrevalence()`, those should be treated as regression bugs.